include/unifex/finally.hpp

```diff
@@ -696,36 +716,50 @@ namespace unifex
                 static_cast<Receiver&&>(r)};
       }
 
+      friend constexpr auto tag_invoke(tag_t<unifex::blocking>, const type& self) noexcept {
+        if constexpr (
```

Comment:
This is non-obvious. Why do you have this `if constexpr`?

Reply:
In the two branches, the return types are different. When the condition is true, we have access to the blocking property at compile time and can return a compile-time constant. (`blocking_kind::constant` is an alias for an instantiation of `std::integral_constant`.) When the condition is false, we fall back to a runtime computation and return the result as a `blocking_kind`, which is a wrapper around an ordinary `enum`.